x/plans/keeper/migrations.go

```diff
@@ -22,3 +25,39 @@ func (m Migrator) Migrate2to3(ctx sdk.Context) error {
 	}
 	return nil
 }
+
+func (m Migrator) Migrate3to4(ctx sdk.Context) error {
+	planIndices := m.keeper.GetAllPlanIndices(ctx)
+
+	for _, planIndex := range planIndices {
+		var oldPlanStruct v3.Plan
+		if found := m.keeper.plansFS.FindEntry(ctx, planIndex, uint64(ctx.BlockHeight()), &oldPlanStruct); !found {
+			return fmt.Errorf("could not find plan with index %s", planIndex)
+		}
+
+		planPolicy := projecttypes.Policy{
+			GeolocationProfile: uint64(1),
+			TotalCuLimit:       oldPlanStruct.ComputeUnits,
+			EpochCuLimit:       oldPlanStruct.ComputeUnitsPerEpoch,
+			MaxProvidersToPair: oldPlanStruct.MaxProvidersToPair,
+		}
+
+		newPlanStruct := types.Plan{
+			Index:                    oldPlanStruct.Index,
+			Block:                    oldPlanStruct.Block,
+			Price:                    oldPlanStruct.Price,
+			OveruseRate:              oldPlanStruct.OveruseRate,
+			AllowOveruse:             oldPlanStruct.AllowOveruse,
+			Description:              oldPlanStruct.Description,
+			Type:                     oldPlanStruct.Type,
+			AnnualDiscountPercentage: oldPlanStruct.AnnualDiscountPercentage,
+			PlanPolicy:               planPolicy,
+		}
+
+		err := m.keeper.plansFS.ModifyEntry(ctx, planIndex, uint64(ctx.BlockHeight()), &newPlanStruct)
```

Comment:
ModifyEntry() should be given the *specific* block of the entry, which we should now (see above).